posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,800 @@
+* Good support of SIMD intrinsics and target-specific builtins.
+* The GCC codegen can now bootstrap rustc.
+* Added support for packed structs.
+* Preparation for a future distribution via rustup with a feature flag allowing to use upstream libgccjit 12.
+* Many bug fixes.
+
+**Regarding new contributors:**
+
+* We got around 5 new contributors.
+* Around 3 stuck around.
+
+**Regarding prioritization and focus:** @antoyo says some importants things we're not doing because we focus on this project is making the rustc_codegen_ssa API more convenient for the backends other than LLVM. That's something they eventually want to work on.
+
+When the GCC backend is good enough and when the Rust for Linux project is ready to start experimenting with it, it might be worth it to help other areas support new architectures for Rust: adding the arch specs to rustc, adding the support for those architectures in some crates like `libc` and `object`.
+
+### Diagnostics Aspirations (👩‍💻)
+
+<!-- https://hackmd.io/@rust-compiler-team/HJ_cgnXd5 -->
+
+**Progress:** The Rust project has made progress on this ambition since the start of the year
+
+**Goals:** we didn’t have “planned goals” for the year, but we did accomplish lots of incremental improvements and started the translation infrastructure which has been in our wishlist for years.
+we think our chosen problem is never-ending and will never be “resolved”;
+there is specific feature and integration work that can be tackled in bounded amount of times, but the improvement of diagnostics is a fractal one, and every time we raise the bar, the expectation of our users increases and our increased experience makes us attempt bolder things.
+
+**How it started:** at start of 2022, we knew some problems existed, but we did not yet have a specific goal in mind for solving all such problems.
+for other problems, we had a plan for how to achieve a specific goal, but we did not have intermediate progress points (aka “milestones”) established.
+
+**How it's going:** we understand the problem better than we did at the start of the year
+we have a high-level end-to-end plan to solve the problem
+we have milestones that represent individual units of work for the near term that each make progress towards a solution
+we have Rust contributors who have agreed to help with the units of work that we have identified.
+Further more, some of our work has reached Rust programmers; in some cases, we do not know if it has improved Rust for them,
+and in other cases, what we learn of their usage is informing our plans going forward
+
+**Details:** wg-diagnostics isn't as structured as it could be, which translates to work being taken on piecemeal by individuals as at their own pace and leisure. This affects the size of projects that we can tackle, *but* still relatively big projects, like the [translation machinery](https://hackmd.io/@davidtwco/rkXSbLg95) were started and have enough momentum to make progress going forward.
```

Comment:
Maybe there's a missing word here, or the "as" is unintended ?

```suggestion
**Details:** wg-diagnostics isn't as structured as it could be, which translates to work being taken on piecemeal by individuals at their own pace and leisure. This affects the size of projects that we can tackle, *but* still relatively big projects, like the [translation machinery](https://hackmd.io/@davidtwco/rkXSbLg95) were started and have enough momentum to make progress going forward.
```